frontend/packages/dev-console/src/components/import/devfile/DevfileImportForm.tsx

```diff
@@ -22,9 +28,49 @@ const DevfileImportForm: React.FC<FormikProps<FormikValues> & DevfileImportFormP
   setFieldValue,
 }) => {
   const { t } = useTranslation();
+  const searchParams = new URLSearchParams(window.location.search);
+  const gitRepoUrl = searchParams.get('gitRepo');
+  const devfileName = searchParams.get('devfileName') || detectGitRepoName(gitRepoUrl);
+  const formType = searchParams.get('formType');
+  const [selectedSample, setSelectedSample] = React.useState<DevfileSample>();
   const [, devfileParseError] = useDefileServer(values, setFieldValue);
   useDevfileDirectoryWatcher(values, setFieldValue);
 
+  React.useEffect(() => {
+    if (gitRepoUrl) {
+      const gitType = detectGitType(gitRepoUrl);
+      const gitService = getGitService({ url: gitRepoUrl }, gitType as GitProvider);
+      setFieldValue('name', createComponentName(devfileName));
+      setFieldValue('git.url', gitRepoUrl);
+      setFieldValue('git.type', gitType);
+      gitService
+        .getDevfileContent()
+        .then((contents) => setFieldValue('devfile.devfileContent', contents))
+        .catch(() => setFieldValue('devfile.devfileHasError', true));
+
+      if (formType === 'sample') {
+        const defaultAppGroup = 'sample-app';
+        setFieldValue('application.initial', defaultAppGroup);
+        setFieldValue('application.name', defaultAppGroup);
+        setFieldValue('application.selectedKey', defaultAppGroup);
+      }
+    }
+  }, [gitRepoUrl, devfileName, formType, setFieldValue]);
+
+  React.useEffect(() => {
+    if (devfileName) {
+      const payload = {
+        registry: 'pkg/devfile/sample-placeholder.json',
+      };
+      coFetchJSON
```

Comment:
This would fetch all the devfile samples every time `devfileName` changes. The list shouldn't change that often so Ideally it should only fetch the list once and search through it.

Reply:
Removed effect dependencies.